src/qt/calctrl.cpp

```diff
@@ -120,6 +125,11 @@ void wxCalendarCtrl::UpdateStyle()
     RefreshHolidays();
 }
 
+bool wxCalendarCtrl::IsQDateValid(const QDate &date) const
+{
+    return date.isValid() && !date.isNull() && date.day() > 0 && date.month() > 0 && date.year() > 0 && date.year() < MAX_YEAR_QT;
```

Comment:
`isValid()` is the same as `isNull()` according to Qt docs, so just one of those is needed. And I'm pretty sure that a valid date can't have neither day nor the month of 0. OTOH I'm not sure why does it need to have a positive year (what if want the user to enter Aristotle's birth date, eh?).

So, all in all, I'd change this line like this:
```suggestion
    return date.isValid() && date.year() < MAX_YEAR_QT;
```

Reply:
The suggestion to change the check here so that: `date.year() > 0` is removed causes the test to fail with the following exception during the call `wxQtConvertDate()`: _date out of range - can't convert to JDN_

I appreciate your suggestion to remove this check in terms of reasonable usage of date pickers, however I believe we will need to keep this. All of your other suggested improvements are valid and are appreciated.

All of this may not be necessary if we instead add `#ifndef __WXQT__` around the corresponding unit test and mentioning that Qt port limits the entry to the dates before 8000-01-01 in the documentation. I will make the suggested changes and add a further commit shortly, then we can make a decision on whether or not we keep these changes or simply do not perform these tests for QT.